test/lifecycles/WrapperLifecycle.test.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2023 Mikhail Aheichyk
+Copyright 2023 Nordeck IT + Consulting GmbH.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+
+import { RuntimeModule } from '../../src/RuntimeModule';
+import { WrapperLifecycle, WrapperListener, WrapperOpts } from '../../src/lifecycles/WrapperLifecycle';
+
+describe('WrapperLifecycle', () => {
+    let module: RuntimeModule;
+
+    beforeAll(() => {
+        module = new class extends RuntimeModule {
+            constructor() {
+                super(undefined as any);
+
+                this.on(WrapperLifecycle.Wrapper, this.wrapperListener);
+            }
+
+            protected wrapperListener: WrapperListener = (wrapperOpts: WrapperOpts) => {
+                wrapperOpts.Wrapper = ({ children }) => {
+                    return <><div>Header</div>{children}<div>Footer</div></>;
+                };
+            };
+        };
+    });
+
+    it('should wrap a matrix client with header and footer', () => {
+        const opts: WrapperOpts = { Wrapper: React.Fragment };
+        module.emit(WrapperLifecycle.Wrapper, opts);
+
+        render(<opts.Wrapper><span>MatrixChat</span></opts.Wrapper>);
+        screen.getByText(/Header/i);
+        screen.getByText(/MatrixChat/i);
+        screen.getByText(/Footer/i);
```

Comment:
I think we should write proper assertions rather than relying on the fact that `screen.getBy` will throw if the item is not found.

```
expect(screen.getByRole("banner")).toBeInTheDocument()
```

And we should rely on aria roles when possible. `header` uses the [`banner` role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/banner_role) by default and `footer` uses the [`contentinfo` role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/contentinfo_role)

We should also add an assertion to make sure that the header/footer are sibling elements to the Matrix chat.

Reply:
Thanks for the comment, sounds all good, I have updated the test